Auth.md

```diff
@@ -0,0 +1,87 @@
+>
+> _To request an API token, you must have access to a valid OIDC access token for your user. You can request one of these with a direct access grant to Keycloak's ODC endpoint:_
+>
+>     curl -L 'https://test.eodatahub.org.uk/keycloak/realms/eodhp/protocol/openid-connect/token' \
+>         -H 'Content-Type: application/x-www-form-urlencoded' \
+>         --d 'client_id=<client_id>' \
+>         --d 'client_secret=<client_secret>' \
+>         --d 'username=<username>' \
+>         --d 'password=<password>' \
+>         --d 'grant_type=password'
+>
+> _With a valid access token, you can request an API token using:_
+>
+>     curl -L -X POST 'https://test.eodatahub.org.uk/api/tokens' \
+>         -H 'Authorization: Bearer <access_token>'
+>
+> _You should take note of the API token as you will not be able to retrieve it again (it will not appear in any GET requests)._
+>
+> _The following endpoints are available to manage your tokens:_
+>
+> - [GET, POST] https://test.eodatahub.org.uk/api/tokens
+> - [GET, DELETE] https://test.eodatahub.org.uk/api/tokens/<token_id>
+
+API tokens act as offline access tokens. They are valid until deleted.
+
+To authenticate with an API token, include an `Authorization: Bearer <api_token>` header with your request.
+
+You can request as many API tokens as you wish.
+
+### Accessing Workspace
+
+Access to workspace contents can be managed both via a browser with a session cookie and through an API token as described in the API Tokens section. Preflighted API requests are also supported.
+Unauthenticated requests will be redirected to Keycloak to sign in. Users may only access files in their own workspace or workspace groups that they are members of.
+
+An example request for accessing a workspace file is as follows:
+
+```http
+GET /files/<bucket_name>/<path_to_file> HTTP/1.1
+Host: <workspace_name>.workspaces.test.eodhp.eco-ke-staging.com
+Authorization: Bearer <api_token>
+```
+
+#### S3 Bucket
+
+Workspace object stores, including saved catalogs and workflow outputs, are accessible at
+`https://<workspace_name>.workspaces.test.eodhp.eco-ke-staging.com/files/<bucket_name>/<path_to_file>`.
+The bucket name and path to workflow outputs may be obtained from ADES outputs.
+
+#### Block Storage
+
+Workspace block stores relevant to AppHub are accessible at
+`https://<workspace_name>.workspaces.dev.eodhp.eco-ke-staging.com/files/workspaces/<path_to_file>`.
+
+## App Developers
+
+Keycloak clients can be set up by EO DataHub developers on behalf of application developers (one per app). Please notify the developers of the OIDC code flows you wish to utilise, as these are enabled on an individual basis.
+
+The options in Keycloak are (OIDC terminology in brackets):
+
+- Standard flow (Authorization Code Flow)
+- Direct access grant (Resource Owner Password Credentials Grant)
+- Implicit flow (Implicit Flow)
+- Service account flow (Client Credentials Grant)
+
```

Comment:
It may be worth adding that the resource owner password credentials grant and implicit flow are both deprecated and should not normally be used. Most applications should use the authorization code flow.